### PR TITLE
Unhack `getAndUpdates`

### PR DESCRIPTION
**Pull request by armanbilge**
Fixes https://github.com/armanbilge/calico/issues/133.

Let me know if this fixes your issue. Sorry for using the cheap hack before 😂

### CODE REVIEW

File: calico/src/main/scala/calico/syntax.scala [hunk 1]
```diff
@@ -65,8 +65,10 @@ extension [F[_], A](fa: F[A])
 
 extension [F[_], A](signal: Signal[F, A])
   private[calico] def getAndUpdates(using Concurrent[F]): Resource[F, (A, Stream[F, A])] =
```
Comment (armanbilge) [inline calico/src/main/scala/calico/syntax.scala:67]: Actually I'd like to upstream this to FS2, but not 100% sure yet.

**Review by 2chilled — APPROVED** [review]
Yep, this fixes it, awesome! 

> Actually I'd like to upstream this to FS2

Good idea. Maybe you should consider renaming it to `getAndDiscrete` to avoid confusion which of both Stream options this is returning.

**Comment by armanbilge** [comment]
> Yep, this fixes it, awesome!

Great, thanks for trying that!

> Maybe you should consider renaming it to `getAndDiscrete` to avoid confusion which of both Stream options this is returning.

The problem is that `getAndDiscrete` sounds like it is calling `signal.get` and then `signal.discrete`. But `discrete` always emits the present value (equivalent to `get`), which is actually what we want to avoid.

The idea here is we should be able to `get` the current value, render the view, and then setup the update subscription _after_ rendering, so that it appears to have a faster UX.

**Comment by 2chilled** [comment]
> sounds like it is calling signal.get and then signal.discrete
Yeah, naming is hard :) `headAndTailDiscrete` comes to my mind. When in doubt, I'd leave it as is then.

**Comment by 2chilled** [comment]
> setup the update subscription after rendering

Yes, I like this idea. But not sure about if it makes a noticeable difference in practice. Have you done some benchmarks on that?

**Comment by armanbilge** [comment]
> But not sure about if it makes a noticeable difference in practice. Have you done some benchmarks on that?

I did some "eyeball" benchmark when I was working on https://github.com/armanbilge/calico/issues/45. It can make a difference when you are creating many subscriptions at the same time. For example, on the TodoMvc when there are 100s of todos.

**Pull request merged**